rtio: Sign compare error during project build with SPI enabled

Describe the bug
After change https://github.com/zephyrproject-rtos/zephyr/pull/59716/files compiling project with SPI enabled and -Werror and -Wsign-compare ends with error in rtio header.

To Reproduce

    In spi_loopback ztest CMakeList file add :
    set(CMAKE_CXX_FLAGS "-Wall -Werror -Wsign-compare")
    set(CMAKE_C_FLAGS "${CMAKE_CXX_FLAGS}")
    Build test for nucleo_wb55rg or other STM

Expected behavior
Build successful.

Logs and console output
In file included from /home/zephyr/include/zephyr/drivers/spi.h:29,
                 from /home/zephyr/tests/drivers/spi/spi_loopback/src/spi.c:17:
/home/zephyr/include/zephyr/rtio/rtio.h: In function 'z_impl_rtio_submit':
/home/zephyr/include/zephyr/rtio/rtio.h:1443:41: error: comparison of integer expressions of different signedness: 'atomic_val_t' {aka 'long int'} and 'uintptr_t' {aka 'long unsigned int'} [-Werror=sign-compare]
 1443 |         while (atomic_get(&r->cq_count) < cq_count) {


Environment:

    OS: Ubuntu 22.04
    Toolchain: Zephyr SDK 16.1
    Commit SHA : https://github.com/zephyrproject-rtos/zephyr/commit/dbea999347128c67a3e06eb34ee67f49be923658

--- a/include/zephyr/rtio/rtio.h
+++ b/include/zephyr/rtio/rtio.h
@@ -1420,7 +1420,7 @@ static inline int z_impl_rtio_submit(struct rtio *r, uint32_t wait_count)
 		r->submit_count = wait_count;
 	}
 #else
-	uintptr_t cq_count = atomic_get(&r->cq_count) + wait_count;
+	uintptr_t cq_count = (uintptr_t)atomic_get(&r->cq_count) + wait_count;
 #endif
 
 	/* Submit the queue to the executor which consumes submissions
@@ -1440,7 +1440,7 @@ static inline int z_impl_rtio_submit(struct rtio *r, uint32_t wait_count)
 			 "semaphore was reset or timed out while waiting on completions!");
 	}
 #else
-	while (atomic_get(&r->cq_count) < cq_count) {
+	while ((uintptr_t)atomic_get(&r->cq_count) < cq_count) {
 		Z_SPIN_DELAY(10);
 		k_yield();
 	}

